earth2studio/data/goes.py

```diff
@@ -0,0 +1,659 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import functools
+import hashlib
+import os
+import pathlib
+import shutil
+from datetime import datetime, timezone
+
+import nest_asyncio
+import numpy as np
+import s3fs
+import xarray as xr
+from loguru import logger
+from tqdm.asyncio import tqdm
+
+from earth2studio.data.utils import (
+    datasource_cache_root,
+    prep_data_inputs,
+)
+from earth2studio.lexicon import GOESLexicon
+from earth2studio.utils.type import TimeArray, VariableArray
+
+
+class GOES:
```

Comment:
Wince you use pyproj probably need to add
https://github.com/NVIDIA/earth2studio/blob/main/earth2studio/data/hrrr.py#L66